src/wrappers/koa/KoaExecutor.ts

```diff
@@ -0,0 +1,80 @@
+import PluginContext from '../../plugins/PluginContext';
+import ExecutionContext from '../../context/ExecutionContext';
+import WrapperUtils from '../WebWrapperUtils';
+import Utils from '../../utils/Utils';
+import {HttpTags, SpanTags, TriggerHeaderTags} from '../../Constants';
+import InvocationSupport from '../../plugins/support/InvocationSupport';
+import ConfigProvider from '../../config/ConfigProvider';
+import ConfigNames from '../../config/ConfigNames';
+
+export function startInvocation(pluginContext: PluginContext, execContext: any) {
+    execContext.invocationData = WrapperUtils.createInvocationData(execContext, pluginContext);
+}
+
+export function finishInvocation(pluginContext: PluginContext, execContext: any) {
+    WrapperUtils.finishInvocationData(execContext, pluginContext);
+}
+
+export function startTrace(pluginContext: PluginContext, execContext: ExecutionContext) {
+    WrapperUtils.startTrace(pluginContext, execContext);
+
+    const {rootSpan, request, response} = execContext;
+
+    const resourceName = request.path || '';
+    const triggerOperationName = request.hostname + resourceName;
+
+    execContext.triggerOperationName = triggerOperationName;
+    execContext.applicationResourceName = resourceName;
+    rootSpan.operationName = resourceName;
+
+    Utils.copyProperties(
+        request,
+        [
+            'method',
+            'query',
+            'hostname',
+            'path',
+        ],
+        rootSpan.tags,
+        [
+            HttpTags.HTTP_METHOD,
+            HttpTags.QUERY_PARAMS,
+            HttpTags.HTTP_HOST,
+            HttpTags.HTTP_PATH,
+        ],
+    );
+
+    InvocationSupport.setAgentTag(HttpTags.HTTP_METHOD, request.method);
+    InvocationSupport.setAgentTag(SpanTags.TRIGGER_OPERATION_NAMES, [triggerOperationName]);
+
+    if (!ConfigProvider.get<boolean>(ConfigNames.THUNDRA_TRACE_REQUEST_SKIP)) {
+        Utils.copyProperties(
+            request,
+            ['body'],
+            rootSpan.tags,
+            [HttpTags.BODY],
+        );
+    }
+
+    if (triggerOperationName) {
+        response.set(TriggerHeaderTags.RESOURCE_NAME, triggerOperationName);
+    }
+}
+
+export function finishTrace(pluginContext: PluginContext, execContext: ExecutionContext) {
+    WrapperUtils.finishTrace(pluginContext, execContext);
+
+    const {rootSpan, response, request} = execContext;
+
+    if (request._matchedRoute) {
```

Comment:
I think, we should not check whether `request._matchedRoute` is exist. Because if it is not exist, the code continues with `request.path || ''` by defauşt